### ISSUE TITLE
"Multiboot buffer is too small." after upgrade to ESXi-6.7.0-20181002001-standard (Build 10302608)

### ISSUE THREAD

**Issue by giro83** [vmware/esx-boot#1]
Hi,

Apologies if this is the wrong place where to report my issue. Feel free to say so should this be the case.

I have a 2014 Apple Mac Mini (the last 2.6GHz Quad-Core Intel Core i7 they made) on which I run ESXi (for home/educational purposes only - no production stuff on it, clearly).

I've run several ESXi versions on it over the years: 5.5, 6.0, 6.5 and now 6.7. It's the perfect device on which to run ESXi at home (except for the price), given it's small, quiet, stylish, etc.

Anyway, today I upgraded from ESXi-6.7.0-20181004001-standard (Build 10176752) to ESXi-6.7.0-20181002001-standard (Build 10302608). The upgrade has essentially "bricked" the device.

ESXi is seen booting and loading files, but after a few seconds returns:

```
Shutting down firmware services...
Multiboot buffer is too small.
Unrecoverable error
```
Also attaching a photo of the screen below.

I searched around, and I couldn't find references to this error on the Internet. However, I have found this project, and it seems the error comes from here:

https://github.com/vmware/esx-boot/blob/abe46c4b2edb507e5486c3996075d9bd7ade4ba1/mboot/mutiboot.c#L154

Again, I'm fully aware my host is not a supported device... but it does the job for me, and I have no reason to replace it yet.

Can I do anything to recover it, short of performing a factory reset of 6.7.0 and not upgrading to this imageprofile or newer?

Is this error related to UEFI boot?

Can I switch to legacy boot?

Can I pass any parameter to the kernel line to increase the buffer which the error refers to?

Thanks in advance,
Stefano

![img_0027](https://user-images.githubusercontent.com/20286572/47267202-62f66580-d538-11e8-943d-0520564ea2a9.JPG)

**Comment by giro83** [comment]
Screen grab when booting with Shift+V:

![img_0047](https://user-images.githubusercontent.com/20286572/47323747-fae97180-d654-11e8-8748-ad44fcac63c9.JPG)

**Comment by andreiw** [comment]
Hi @giro83, thanks for the report. There's one thing we could try, but need to get access to a UEFI shell. Unfortunately, this is a Mac we're talking about - you wouldn't happen to have rEFInd or rEFIt installed, would you?

**Comment by giro83** [comment]
Hi @andreiw, I can boot the Mac Mini with a (e.g.) Gentoo Live USB stick. I can get networking going and install packages needed (probably including the ones you mentioned). I should then be able to execute commands / follow a procedure.

https://wiki.gentoo.org/wiki/Refind

**Comment by andreiw** [comment]
Okay, here's how I could get boot to the UEFI shell on a mac.

This should work on any OS, but I will use OS X here.

1) Download http://downloads.sourceforge.net/refit/refit-bin-0.14.tar.gz
2) Uncompress to a FAT32 formatted USB key
3) Rename "\efi\refit\" to "\efi\boot\"
4) Rename "\efi\boot\refit.efi" to "\efi\boot\bootx64.efi"

Reboot, hold "Option key", select "UEFI shell" (it has a terminal icon).

Once you have the Shell running, find the \efi\boot\bootx64.efi file corresponding to the ESX installation. Use "map" command to print the available UEFI equivalents of DOS "drive letters", and 
"cd" and "dir" to navigate and list. E.g.:

    Shell> fs5:
    Shell> cd efi\boot\
    Shell> dir

Once you find it, run it with the "-U" option:

    Shell> bootx64.efi -U

**Comment by andreiw** [comment]
Hint: there will be a lot of output inside "map". To pause while scrolling run it like this, with the "-b" argument:

     Shell> map -r -b

**Comment by andreiw** [comment]
Just a visual example of what the USB stick directory structure looks like, compared to the refit TAR file:

Top is USB stick. Bottom is tar file.

![image](https://user-images.githubusercontent.com/732523/47368586-eb4b4680-d6af-11e8-8989-5cb5a825f47a.png)

**Comment by giro83** [comment]
@andreiw Success! You have totally resurrected my ESXi installation :)

If I understood correctly what I just did, this seems to have (permanently?) disabled UEFI. I have verified that the host can now boot ESXi (as before) completely unattended.

Do I need to worry about future updates re-enabling this for me?

Will you be doing anything at your end?

Anyway, thank you very much so far.

**Comment by giro83** [comment]
Mine also seems to be back at build 10176752, however. Should I attempt the upgrade again?

**Comment by andreiw** [comment]
Hi @giro83. Thanks for doing the steps to help triage the issue. This option boots without further EFI support in ESX (for the current boot, not permanently). We'll need to investigate this further internally to diagnose the actual cause.

I do want to point out that this Github issue tracker is not the right channel to raise customer support issues or get support. Our communication here, as useful as it is to get data, does not constitute any acknowledgement/receipt of a support issue, or provide a guarantee/promise of remediation. Please raise a support request via https://www.vmware.com/support/file-sr.html, and you should contact your TAM if your have one. Filing an SR is the way forward to report and address product issues and defects (e.g. @quotesbnbi). 

Thanks for your help in identifying the symptoms and the affected systems, and thanks for the heads up. I'll close this as there is nothing else to be done via this (unofficial) channel.

**Comment by andreiw** [comment]
@giro83 and @quotesbnbi, would it be possible to get more system configuration information on the machines you've seen the problem with?

- Make/year/model of machine. E.g. MacBook Pro (Retina, 15-inch, Mid 2015), MacBookPro11,5.
- How much RAM the system has.
- Boot ROM and SMC versions, if possible (e.g. MBP114.0172.B25 and 2.30f2).

**Comment by NM-IT** [comment]
Just so you know you're not alone: We also got stuck with this problem this weekend while (trying to) upgrade to 6.7U1 from 6.7.0, 10176752. We tried a complete reinstall from Mware-VMvisor-Installer-6.7.0.update01-10302608.x86_64.iso - but after reboot got stuck just the same.
We run 3 x MacPro6.1 hosts (Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz) with 128GB RAM (from OWC).

**Comment by giro83** [comment]
> Just so you know you're not alone: We also got stuck with this problem this weekend while (trying to) upgrade to 6.7U1 from 6.7.0, 10176752. We also tried a complete reinstall from VMware-VMvisor-Installer-6.7.0.update01-10302608.x86_64.iso - but after reboot got stuck just the same.
> We run 3 x MacPro6.1 hosts (Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz) with 128GB RAM (from OWC).

This is interesting/useful information, because although my host is unsupported, the Mac Pro *is* supported according to https://www.vmware.com/resources/compatibility/search.php

**Comment by NM-IT** [comment]
Yes it is supported - so this really bugs me. Unfortunately my machines are in production (offsite@colocation center) - so I can't really dig into troubleshooting this more at this point.

**Comment by giro83** [comment]
But @andreiw just to be clear, I totally appreciate what you've done for me (outside of any official support channel). It's more than I expected when I opened this issue, and I'm really happy.

I will try to figure out on my own why my ESXi installation is now booting correctly unattended (albeit from the old version). Maybe there was more than one ESXi installation to boot from and I picked the wrong one, I don't know.

I will also supply the info you requested ASAP.

**Comment by quotesbnbi** [comment]
We run the following hardware:
MacPro6,1, Late 2013, A1481
32GB memory
12 core processor: 12 CPUs x Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
Bios: MP61.88Z.0120.B00.1708080652
Bios release date: Monday, August 07, 2017, 20:00:00 -0400

**Comment by lamw** [comment]
**Disclaimer:** VMware Employee

Thanks @andreiw for forwarding this internally. We're trying to see if we can reproduce this problem with Engr. We definitely like to get as much information as possible (configuration and how the USB Installer was created) to make sure we're running the exact same workflow. 

@NM-IT The Mac Pro 6,1 is indeed a supported platform and thanks for filing an SR (this is the best way to engage with VMware, so our support team can work with you). Would you mind sharing the SR number?

@quotesbnbi Thanks for the info, I'll share this back with the Engr team

**Comment by NM-IT** [comment]
VMware Support Request 18963226910 is the case number used here.

**Comment by dminion909** [comment]
Hi everyone. We are also having this issue. Not just with the 6.7 u1 but with all previous versions as well. We are running ESXi on quite a few MacPro 6,1 systems. This system in particular has a newer firmware version which we believe is the culprit. This is the first version of ESXi I've actually managed to get installed on this system. Every other version has had the "nfs41client failed to load" followed by no network adapters issue as previously experienced by @lamw. I am hesitant to try to flash the ROM back to a previous version as turning this system into a brick would be a pretty costly mistake. If you guys have any ideas on how to get 6.7 or 6.0 to install on this MP6,1, I'd really appreciate it.

**Comment by lamw** [comment]
@dminion909 Can you share the version of the firmware that you're seeing for the newer system? And a few of firmware version for systems without issues (unless they're all the same). Thanks

**Comment by dminion909** [comment]
I'm not sure of the firmware on the ones in production but they are most likely the original firmware that was installed when the units were shipped. ESX was immediately installed on them so they never had a chance to update.

This one had El Capitan installed and has updated. I managed to get ESXi 6.7 u1 HPE installed to a USB flash drive. The regular 6.7 u1 wouldn't install as it would get the Multiboot error on install. I started this system up again after getting ESXi installed and on second boot, I'm getting:
VMware Hypervisor Recovery
BANK5: invalid configuration.
BANK6: invalid configuration.
Ho hypervisor found.

I can boot from Mac OS X recovery and get the firmware of this one. It will take me a bit so I'll get it for you ASAP. Thanks @lamw

**Comment by dminion909** [comment]
Here you go @lamw. This is the system I'm not able to get working:

Model Name: Mac Pro
Model Identifier: MacPro6,1
Processor Name: 8-Core Intel Xeon E5
Processor Speed: 3 GHz
Number of Processors: 1
Total Number of Cores: 8
L2 Cache (per core): 256 KB
L3 Cache: 25 MB
Memory: 64 GB
Boot ROM Version: MP61.0116.B17
SMC Version (system): 2.20f18
Illumination Version: 1.4a6

**Comment by dminion909** [comment]
Update:

I was able to successfully install HPE Customized Image ESXi 6.7.0 Update 1 version 670.U1.10.3.5 released on October 2018 and based on ESXi 6.7.0 Update 1 Vmkernel Release Build 10302608
Installing to the local disk would not work but installing to a 16 GB flash drive worked like a charm... so far. Go figure.
6.7 is not actually the version we need though. We need version 6.0 so I'm going to try that now.

**Comment by jjas0n** [comment]
Chiming in - Have a few Late 2012 Mac Minis. All are exhibiting the issue here. rEFIt got me back online for now. I'd love to get to the bottom of thus.

Thank you!

**Comment by dminion909** [comment]
@jjas0n could you provide some info on what you did to get it working using rEFIt? Perhaps I could use it to fix the issue I'm running into with installing 6.0...
Thanks!

**Comment by jjas0n** [comment]
@dminion909 - I just followed the instructions from @andreiw above to boot from the VMWARE bootx64.efi loader with the -U flag. That’s it. 

I cannot apply the 6.7.0U1 update at this point, but at least I’m running. 

Hope this helps.

**Comment by dminion909** [comment]
Morning all. I was able to get ESXi 6.0.0 4192238 installed to the local drive on MacPro6,1. The way I was able to do this was by booting from an El Capitan USB installer, launching the Disk Utility and wiping all the partitions on the system disk. Once I did that, I booted from the ESXi USB installer and the installation ran just fine. No nfs41client issues. No issues finding the NICs. No Multiboot errors. It successfully pulled an IP and I can log in through vSphere. Haven't migrated VMs or joined the cluster yet but this is as far as I've gotten. This might fix your issues for 6.7 as well.

We also have a MacMini7,1 that wouldn't install and this worked on that system as well.

Specs:
Model Name: Mac Pro
Model Identifier: MacPro6,1
Processor Name: 8-Core Intel Xeon E5
Processor Speed: 3 GHz
Number of Processors: 1
Total Number of Cores: 8
L2 Cache (per core): 256 KB
L3 Cache: 25 MB
Memory: 64 GB
Boot ROM Version: MP61.0116.B17
SMC Version (system): 2.20f18
Illumination Version: 1.4a6

Hope that helps. Have a great weekend everyone 😄 
@lamw @andreiw @jjas0n @giro83

**Comment by mightyfree** [comment]
I experienced this issue on a 2012 Mac Mini 6,2 hosted at Mac Stadium, upgrading from 6.7 10176752 to 6.7U1, which was unrecoverable. Remediation included installing ESXi 6.5 over existing 6.7 install using 6.5 ESXi installation media, while ensuring the option to preserve existing VMs was checked. Remote hands were needed to make this happen. The Mac mini was subsequently upgraded up to 6.7 10176879 successfully, while the 6.7U1 update is investigated and fixed.

**Comment by mightyfree** [comment]
@dminion909 Mac Stadium (macstadium.com) hosts hundreds of Mac Pros, many with a variety of flavors of ESXi installed. You may try reaching out to their tech support team for ideas.

**Comment by dminion909** [comment]
Thanks @mightyfree . I actually did early last week and they haven't had this particular issue. I'm glad you reminded me though... I'll email them back with an update.

**Comment by jorge-dbx** [comment]
Experiencing this issue as well on about a dozen of Mac Pros (the earliest ones we purchased for our fleet).  
Mac Pro Boot Rom: MP61.0116.B16 (with issues)

Fixed it by installing MacOS High Sierra and connecting it to the network to download all the latest firmware updates.

Post-install Mac Pro Boot rom: MP61.0120.B00

Open case with VMware: 18970432210

@lamw hope this helps

**Comment by s0ftice** [comment]
Today's ESXi-6.7.0-20181104001-standard (Build 10764712) release did not solve it for me unfortunately...

**Comment by erpomik** [comment]
FYI: I tried to upgrade/re-install two Mac Pro's this weekend.
Got "Multiboot buffer is too small" on both.

**Comment by lamw** [comment]
FYI - https://www.virtuallyghetto.com/2018/11/heads-up-apple-mac-pro-61-fails-to-boot-after-upgrading-to-esxi-6-7-update-1.html <-Includes VMware KB, which folks can follow to hopefully help them revert back to previous ESXi installation. You can also subscribe to the KB for updates regarding this issue

**Comment by mightyfree** [comment]
As @jorge-dbx stated above, it appears the issue may be due to outdated boot rom on our Mac hardware. When MacOS is installed, these updates are installed as MacOS is regularly updated.

But because our hardware has ESXi installed - sometimes for years at a time, we're missing out on these important boot rom updates.

While it's possible for VMware to code a workaround, perhaps the fix is as easy as updating the boot roms per @jorge-dbx 's instructions. I'll look into this potential resolution on a 2012 Mac mini that is experiencing this error, and report back.

**Comment by quotesbnbi** [comment]
As noted earlier, I am also running the same bios as @jorge-dbx and am unable to run 6.7u1.  I wiped the mac pro and installed every OS up to and including high sierra in Nov 2017 in order to get all firmware updates.  Unless a newer update has come out since then that isn't being reported by @jorge-dbx, then something else is going on here.  In any case, it sounds like the esxi-boot code has been fixed and we are just awaiting a release.   Thought I should mention it before people start blowing away their esx data to install firmware that isn't certain to fix the issue.

Jorge's 'boot rom': MP61.0120.B00
Mine as reported by the vmware host web interface: Bios: MP61.88Z.0120.B00.1708080652

**Comment by NM-IT** [comment]
WMware have now responded to my support request 18963226910 (regarding "Shutting down firmware services... Multiboot buffer is too small. Unrecoverable error".
They say: "Kindly note that the issue was resolved in a new release. You can now successfully upgrade to  vSphere 6.7 U2."
I haven't tried it yet - and I haven't even found the update... Anyone see it?

**Comment by shantur** [comment]
@NM-IT : Ask the build number. 
Then check the build patch here : https://my.vmware.com/group/vmware/patch#search

**Comment by TimothyPMann** [comment]
That was an incorrect response, as 67u2 won't be released until early 2019.
We are still hoping to get a fix available before that, but I don't know
when/if that will happen yet. :-(

**Comment by shantur** [comment]
Maybe a patch fix then

**Comment by LordAttilas** [comment]
> Okay, here's how I could get boot to the UEFI shell on a mac.
> 
> This should work on any OS, but I will use OS X here.
> 
> 1. Download http://downloads.sourceforge.net/refit/refit-bin-0.14.tar.gz
> 2. Uncompress to a FAT32 formatted USB key
> 3. Rename "\efi\refit" to "\efi\boot"
> 4. Rename "\efi\boot\refit.efi" to "\efi\boot\bootx64.efi"
> 
> Reboot, hold "Option key", select "UEFI shell" (it has a terminal icon).
> 
> Once you have the Shell running, find the \efi\boot\bootx64.efi file corresponding to the ESX installation. Use "map" command to print the available UEFI equivalents of DOS "drive letters", and
> "cd" and "dir" to navigate and list. E.g.:
> 
> ```
> Shell> fs5:
> Shell> cd efi\boot\
> Shell> dir
> ```
> Once you find it, run it with the "-U" option:
> 
> ```
> Shell> bootx64.efi -U 
> ```

That worked for me for a MacMini 5,2. Big thanks :)

**Comment by jjas0n** [comment]
### **Executive Summary - Its Alive!**
- I have a 4-node Vsphere ESXi 6.7.0U1 fully-patched cluster up and running on Mac Mini Late 2012 (Macmini6,2) hardware.
- New firmware is included in macOS 10.14.2.

**The Idea:**
- macOS includes the firmware updates in the OS updates. 
- I wonder if there is new firmware in 10.14.2? Hmmm...

**The Details Of The Testing:**
- Downloaded the 6.7.0 and 6.7.0U1 installers and put them onto USB keys.
- Downloaded the latest 10.14.2 installer and put it onto a USB key
- Did a test install of 6.7.0U1 from USB to a Late 2012 Mac Mini - and it failed with the expected "Mutiboot buffer is too small"
  - Confirmed failure - as expected.
- Installed 10.14.2 onto the Mac Mini and checked the firmware level.
  - AHA! That took me to a NEW firmware level: 274.0.0.0.0 
- Did a second test install of 6.7.0U1 from USB to a Late 2012 Mac Mini --> and it worked! 
- Repeated this across all Mac Mini nodes successfully.

**Open Items:**
- There has got to be a faster way to apply Apple firmware updates. Anyone know of a better way?

**Web Sites that were helpful along the way:**
- Apple - Mojave (10.14) Hardware Compatibility List
  - https://support.apple.com/en-us/HT201475
- Which EFI firmware should your Mac be using? (version 3) – The Eclectic Light Company
  - https://eclecticlight.co/2018/10/31/which-efi-firmware-should-your-mac-be-using-version-3/
- How to Make a Bootable MacOS Mojave Installer Drive
  - http://osxdaily.com/2018/09/26/make-macos-mojave-boot-usb-installer/
- How to Format USB Flash Drive to FAT32 in Mac OS
  - https://www.admfactory.com/how-to-format-usb-flash-drive-to-fat32-in-mac-os/
- Creating a Bootable USB from ISO with UNetbootin
  - https://sflanders.net/2015/03/26/creating-a-bootable-usb-from-iso-with-unetbootin/

**Comment by giro83** [comment]
>     * There has got to be a faster way to apply Apple firmware updates. Anyone know of a better way?

I, too, would like to avoid performing a full macOS installation to workaround this (I'd have to backup the full datastore, etc.).

I know the macOS installer can be used to launch a terminal, and that can be used to open packages, run things, etc. So if the new firmware comes in a package of some sort and we can get our hands on it...

**Comment by quotesbnbi** [comment]
> > I, too, would like to avoid performing a full macOS installation to workaround this (I'd have to backup the full datastore, etc.).
> 
> I know the macOS installer can be used to launch a terminal, and that can be used to open packages, run things, etc. So if the new firmware comes in a package of some sort and we can get our hands on it...

This ^^^^

**Comment by giro83** [comment]
I plan to give this a go when I have a moment... but if anyone wants to beat me to it:

```
xxxx@XXXX:/ $ cd /Applications/Install\ macOS\ Mojave.app/Contents/SharedSupport/
xxxx@XXXX:/Applications/Install macOS Mojave.app/Contents/SharedSupport $ hdiutil mount InstallESD.dmg 
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified   CRC32 $1A9AFB2A
Checksumming GPT Header (Primary GPT Header : 1)…
 GPT Header (Primary GPT Header : 1): verified   CRC32 $BCB81BA0
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified   CRC32 $900AB018
Checksumming  (Apple_Free : 3)…
                    (Apple_Free : 3): verified   CRC32 $00000000
Checksumming EFI System Partition (C12A7328-F81F-11D2-BA4B-00A0C93EC93B : 4)…
..........
EFI System Partition (C12A7328-F81F-: verified   CRC32 $B54B659C
Checksumming disk image (Apple_HFS : 5)…
..................................................................................................................................................................................................................................................................................................................................................................
          disk image (Apple_HFS : 5): verified   CRC32 $C1140413
Checksumming  (Apple_Free : 6)…
........................................................................................................................................................................................................................................................................................................................................................................
                    (Apple_Free : 6): verified   CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 7)…
GPT Partition Data (Backup GPT Table: verified   CRC32 $900AB018
Checksumming GPT Header (Backup GPT Header : 8)…
  GPT Header (Backup GPT Header : 8): verified   CRC32 $8E6E527F
verified   CRC32 $BBA6988E
/dev/disk2          	GUID_partition_scheme          	
/dev/disk2s1        	EFI                            	
/dev/disk2s2        	Apple_HFS                      	/Volumes/InstallESD
xxxx@XXXX:/Applications/Install macOS Mojave.app/Contents/SharedSupport $ cd /Volumes/InstallESD/
xxxx@XXXX:/Volumes/InstallESD $ find . -iname '*firmware*'
./Packages/EmbeddedOSFirmware.pkg
./Packages/FirmwareUpdate.pkg
xxxx@XXXX:/Volumes/InstallESD $
```

Where that's a copy of macOS Mojave 10.14.2 downloaded via the App Store. Maybe installing one of those two packages from the live environment I described earlier is sufficient.

**Comment by SurinameClubcard** [comment]
> 
> 
> Okay, here's how I could get boot to the UEFI shell on a mac.
> 
> This should work on any OS, but I will use OS X here.
> 
>     1. Download http://downloads.sourceforge.net/refit/refit-bin-0.14.tar.gz
> 
>     2. Uncompress to a FAT32 formatted USB key
> 
>     3. Rename "\efi\refit" to "\efi\boot"
> 
>     4. Rename "\efi\boot\refit.efi" to "\efi\boot\bootx64.efi"
> 
> 
> Reboot, hold "Option key", select "UEFI shell" (it has a terminal icon).
> 
> Once you have the Shell running, find the \efi\boot\bootx64.efi file corresponding to the ESX installation. Use "map" command to print the available UEFI equivalents of DOS "drive letters", and
> "cd" and "dir" to navigate and list. E.g.:
> 
> ```
> Shell> fs5:
> Shell> cd efi\boot\
> Shell> dir
> ```
> 
> Once you find it, run it with the "-U" option:
> 
> ```
> Shell> bootx64.efi -U 
> ```

I tried this procedure on a MacMini5,3 but it didn't work. I reinstalled with ESXi 6.7 Build 8169922 and updated to U1 only to find out that the problem reoccurred. I do believe I had U1 installed before. So now I'm on Build 8169922 stock. Any ideas?

**Comment by corrieb** [comment]
Update - I just ran into this issue with my Mac Minis. The PRAM reset suggested by a contributor to the virtuallyghetto post totally worked. Really simple fix - just hold down option-command-p-r on boot, then when it restarts go through the normal procedure for upgrade.

**Comment by quotesbnbi** [comment]
I tried corrieb's suggestion on a MacPro6,1, booting off of the internal esxi install from the SSD and trying to perform the upgrade immediately after resetting the PRAM.  No difference in behavior.  I installed MacOS 10.14 to USB drive on the pro, then ran the combo updater to 10.14.3 on the usb drive on the pro.  No new firmware for the Pro boot rom.  No change in behavior in the update 1 process.

Per "specter345" here: https://www.virtuallyghetto.com/2018/11/heads-up-apple-mac-pro-61-fails-to-boot-after-upgrading-to-esxi-6-7-update-1.html#comment-55968
I tried adding "preferVmklinux=TRUE" to the "kernelopt=" line of /bootbank/boot.cfg and installing the update.  Update 1 took as well as the December and January patches.  I'm leaving the line there until VMware publishes an official fix.

**Comment by mo8Zomo0** [comment]
I have a MacMini 5,3 (2011 Server with i7).
For me the ESXi-6.7.0-20190104001-standard update seem to have addressed the boot issue with 6.7.
The first reboot did not work. After that I tried to zap the pram without changes, but then I 
booted again with the UEFI -U workaround and I noticed some thing like "updating bootloader" 
during esxi startup and two test restarts afterwards (incl. one with a full shutdown) worked like a charm.

**Comment by flodejr** [comment]
> I have a MacMini 5,3 (2011 Server with i7).
> For me the ESXi-6.7.0-20190104001-standard update seem to have addressed the boot issue with 6.7.
> The first reboot did not work. After that I tried to zap the pram without changes, but then I
> booted again with the UEFI -U workaround and I noticed some thing like "updating bootloader"
> during esxi startup and two test restarts afterwards (incl. one with a full shutdown) worked like a charm.

Are you sure that you have not been reverted back to Build 10176752?  I have the same setup, but each time I do the bootx64.EFI -U, it will revert back to Build 10176752, I can't seem to get to Build 11675023.

**Comment by surzn** [comment]
I met this same issue on my MacMini 6.1(late 2012). Initially I used PRAM reset to restore back to previous working version. And the final resolution I used is to install Mac OS Mojave that upgraded the EFI/BIOS boot version to 274.0.0.0.0. 

After spending hours in installation Mac OS in different ways, finally I figured out that this EFI/BIOS version upgrade will only happen when you do a Mac OS installation in Mac's internal disk, that means installing new Mac OS on external USB key/disk will not upgrade this EFI/BIOS.

By this chance at last, I upgraded SSD to my MacMini, installed Mac OS Mojave, reinstalled ESXi 6.7 update 1 directly, restored my vSphere settings by host profile in VCSA, and copied all my VMs back from HDD to SSD.

I hope this will save you some time without further ado.

**Comment by lamw** [comment]
Hi Folks,

It's been awhile and I'd like to give everyone an update regarding this issue. First of all, thank you very much for your patience. 

The fix for this issue will be part of the upcoming vSphere 6.7 Update 2 release (which was just announced today, see https://blogs.vmware.com/vsphere/2019/04/announcing-vsphere-6-7-update-2-vsphere-platinum-updates-and-vsphere-robo-enterprise.html for more details). To be clear, vSphere 6.7 Update 2 has NOT GA'ed yet, this will happen at a later time and I'll be more than welcome to share the news when that is. When ESXi 6.7 Update 2 is released, the HCL will be updated to reflect support for the Mac Pro 6,1

One final note, I did have another customer share with me his recent experience with using the latest bootrom (129.0.0.0) that was shipped with MacOS Mojave 10.14.14. which *seems* to have resolved the issue for him and allowing him to use vSphere 6.7 Update 1. I don't have any other confirmations, but wanted to be transparent with the information. If you prefer official support, I'd recommend waiting for 6.7 Update 2

**Comment by MacITC** [comment]
Today I updated a MacPro Black with Firmware MP61.88Z.0124.B00.1804111223 from ESXi 6.7 to 6.7U2 remotely via ssh and esxcli commands and indeed the MacPro booted without hanging in the "Multiboot buffer is too small" loop.

1. You will need these commands, nicely illustrated by rtrouton:
https://derflounder.wordpress.com/2018/05/15/upgrading-from-esxi-6-5-to-esxi-6-7-via-ssh-and-esxcli/

2. You will need the exact filename to identify the 6.7U2 update in the repository list:
ESXi-6.7.0-20190402001-standard  = 2019-04-11 (Update 2) (Build 13006603) 
Details can be found on this excellent site, VMware ESXi Patch Tracker:
https://esxi-patches.v-front.de/ESXi-6.7.0.html

3. Once the update command has been completed which you will notice when your terminal window has returned to the command prompt (takes quite a while)

4. Then you simply can type "reboot" and hit return

5. My MacPro Black was indeed rebooted and after about a minute pingable again

6. About 30 seconds after being pingable, login WebGUIwas available too

7. Please keep in mind that after a reboot ssh access is automatically disabled, so you first will have to re-enable it in the ESXi WebGUI if you want to connect via ssh again.

**Comment by mightyfree** [comment]
**### Successful Update to 6.7 U2 - 2012 MacMini Server Edition**

Previous multiboot buffer issues resolved, successful update to 6.7 U2 with no faults.

Great work @lamw and the VMware team, and props for all who contributed in the thread for this issue.

**Comment by SurinameClubcard** [comment]
### Successful Update to 6.7 U2 - 2011 MacMini Server Edition

Specs: MacMini5,3 (Mid-2011 - MC936LL/A)
